org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/commands/OrganizeImportsCommand.java

```diff
@@ -0,0 +1,174 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Microsoft Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Microsoft Corporation - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.jdt.ls.core.internal.commands;
+
+import java.io.File;
+import java.net.URI;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.jdt.core.ICompilationUnit;
+import org.eclipse.jdt.core.IJavaElement;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.core.IPackageFragment;
+import org.eclipse.jdt.core.IPackageFragmentRoot;
+import org.eclipse.jdt.core.JavaModelException;
+import org.eclipse.jdt.core.dom.CompilationUnit;
+import org.eclipse.jdt.ls.core.internal.ImportTextEditConverter;
+import org.eclipse.jdt.ls.core.internal.JDTUtils;
+import org.eclipse.jdt.ls.core.internal.JavaLanguageServerPlugin;
+import org.eclipse.jdt.ls.core.internal.TextEditConverter;
+import org.eclipse.jdt.ls.core.internal.corext.codemanipulation.OrganizeImportsOperation;
+import org.eclipse.jdt.ls.core.internal.corrections.InnovationContext;
+import org.eclipse.jdt.ls.core.internal.corrections.proposals.CUCorrectionProposal;
+import org.eclipse.jdt.ls.core.internal.corrections.proposals.IProposalRelevance;
+import org.eclipse.jface.text.IDocument;
+import org.eclipse.lsp4j.WorkspaceEdit;
+import org.eclipse.ltk.core.refactoring.TextChange;
+import org.eclipse.text.edits.TextEdit;
+
+public class OrganizeImportsCommand {
+
+	public static Object organizeImports(List<Object> arguments) throws CoreException {
+		String fileUri = (String) arguments.get(0);
+		OrganizeImportsCommand command = new OrganizeImportsCommand();
+
+		URI uri = URI.create(fileUri);
+		File file = new File(uri.getPath());
+
+		boolean exists = file.exists();
+		if (!exists) {
+			return new WorkspaceEdit();
+		}
+		if (file.isDirectory()) {
+			return command.organizeImportsInDirectory(fileUri);
+		} else {
+			return command.organizeImportsInFile(fileUri);
+		}
+	}
+
+	public WorkspaceEdit organizeImportsInDirectory(String folderUri) throws CoreException {

```

Comment:
`organizeImportsInDirectory(URI folderUri)`

does it need to be public?

Reply:
Yes, I make it as public for unittest.